modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -1533,6 +1535,7 @@ private GridCacheContext createCache(CacheConfiguration<?, ?> cfg,
 
         boolean nearEnabled = GridCacheUtils.isNearEnabled(cfg);
 
+        CacheCompressionManager compressMgr = new CacheCompressionManager();
```

Comment:
Add a check to fail or warn if compression is configured for an in-memory cache

Reply:
Done.